### PR TITLE
Include super admins to the users list in Exclude settings

### PR DESCRIPTION
**Pull request by marcin-lawrowski**
Fixes https://github.com/xwp/stream/issues/829

It's a work in progress

### CODE REVIEW

File: classes/class-settings.php [hunk 1]
```diff
@@ -114,12 +114,29 @@ public function get_users() {
 		if ( 0 === $users->get_total() ) {
 			wp_send_json_error( $response );
 		}
+		$users_array = $users->results;
+
+		if ( is_multisite() && is_super_admin() ) {
+			$super_admins = get_super_admins();
+			foreach ( $super_admins as $admin ) {
+				$user = get_user_by ( 'login', $admin );
+				$users_array[] = $user;
```
Comment (lukecarbis) [inline classes/class-settings.php]: Should we check if the user already exists in the array first?

File: classes/class-settings.php [hunk 2]
```diff
@@ -114,12 +114,29 @@ public function get_users() {
...
 		$response->message = '';
 		$response->users   = array();
+		$users_added_to_response = array();
+
+		foreach ( $users_array as $key => $user ) {
+			// exclude duplications:
+			if ( array_key_exists( $user->ID, $users_added_to_response ) ) {
+				continue;
+			} else {
+				$users_added_to_response[$user->ID] = true;
```
Comment (lukecarbis) [inline classes/class-settings.php]: Spacing around the variable inside the square brackets.

File: classes/class-settings.php (hunk 1 above)
Comment (marcin-lawrowski) [inline classes/class-settings.php]: I choose to check for duplicates further in the foreach loop.

File: classes/class-settings.php (hunk 2 above)
Comment (marcin-lawrowski) [inline classes/class-settings.php]: Fixed

**Comment by lukecarbis** [comment]
@chacha I've made a few changes in f4a600d. Could you please review, and if you're happy, merge?

File: classes/class-list-table.php [hunk 3]
```diff
@@ -464,10 +464,14 @@ function( $user_id ) {
 			);
 
 			if ( is_multisite() && is_super_admin() ) {
-				$super_admins = get_super_admins();
-				foreach ( $super_admins as $admin ) {
-					$users[] = get_user_by( 'login', $admin );
```
Comment (lukecarbis) [inline classes/class-list-table.php]: This array is an array of `Author`s, not `WP_User`s. Hence the change.

File: classes/class-list-table.php [hunk 4]
```diff
@@ -464,10 +464,14 @@ function( $user_id ) {
...
-				}
+				$super_admins = array_map(
+					function( $login ) {
+						$user = get_user_by( 'login', $login );
+						return new Author( $user->ID );
+					},
+					get_super_admins()
+				);
+				$users = array_unique( array_merge( $users, $super_admins ) );
```
Comment (lukecarbis) [inline classes/class-list-table.php]: Array unique so we don't have duplicates if a Super Admin is also an Administrator for the site.

**Comment by chacha** [comment]
Tested. Good Job @marcin-lawrowski! :+1:

**Pull request merged**